nimble/controller/src/ble_ll_sync.c

```diff
@@ -996,14 +1006,72 @@ ble_ll_sync_check_failed(struct ble_ll_sync_sm *sm)
     ble_ll_sync_est_event_failed(BLE_ERR_CONN_ESTABLISHMENT);
 }
 
+static bool
+ble_ll_sync_check_acad(struct ble_ll_sync_sm *sm,
+                       const uint8_t *acad, uint8_t acad_len)
+{
+    struct ble_ll_acad_channel_map_update_ind *chmu;
+    uint8_t ad_len;
+    uint8_t ad_type;
+
+    /* assume no empty fields */
+    while (acad_len > 2) {
+        ad_len = acad[0];
+        ad_type = acad[1];
+
+        acad += 2;
+        acad_len -= 2;
+
+        if (ad_len == 0 || ad_len > acad_len) {
```

Comment:
`acad_len` is now without `ad_len` and `ad_type`
`ad_len` is with `ad_type`

so `ad_len - 1` shall be no greater than `acad_len`, thus should be `ad_len - 1 > acad_len` here